src/strategies/MangroveOrder.sol

```diff
@@ -120,17 +141,17 @@ contract MangroveOrder is Forwarder, IOrderLogic {
     // * (NAT_THIS+`msg.value`+`res.bounty`, OUT_THIS+`res.takerGot`, IN_THIS+`tko.takerGives`-`res.takerGave`)
 
     bool isComplete = checkCompleteness(tko, res);
-    // when `!restingOrder` this implements FOK. When `restingOrder` the `postRestingOrder` function reverts if resting order fails to be posted amd `partialFillNotAllowed`.
-    // therefore we require `partialFillNotAllowed => (isComplete \/ restingOrder)`
-    require(!tko.partialFillNotAllowed || isComplete || tko.restingOrder, "mgvOrder/mo/noPartialFill");
+    // when `!restingOrder` this implements FOK. When `restingOrder` the `postRestingOrder` function reverts if resting order fails to be posted and `fillOrKill`.
+    // therefore we require `fillOrKill => (isComplete \/ restingOrder)`
+    require(!tko.fillOrKill || isComplete || tko.restingOrder, "mgvOrder/partialFill");
 
     // sending inbound tokens to `msg.sender`'s reserve and sending back remaining outbound tokens
     if (res.takerGot > 0) {
-      router().push(tko.outbound_tkn, callerReserve, res.takerGot);
+      require(router().push(tko.outbound_tkn, callerReserve, res.takerGot) == res.takerGot, "mgvOrder/pushFailed");
     }
-    uint inbound_left = tko.takerGives - res.takerGave;
-    if (inbound_left > 0) {
-      router().push(tko.inbound_tkn, callerReserve, inbound_left);
+    uint inboundLeft = tko.takerGives - res.takerGave;
+    if (inboundLeft > 0) {
+      require(router().push(tko.inbound_tkn, callerReserve, inboundLeft) == inboundLeft, "mgvOrder/pushFailed");
```

Comment:
In the specific case of `SimpleRouter` push either returns `amount` or throws. To protect against future change we added checks of the returned amount.